src/API.jl

```diff
@@ -948,11 +955,17 @@ function precompile(ctx::Context)
                     return
                 end
                 try
-                    Base.compilecache(pkg, sourcepath)
                     was_recompiled[pkg.uuid] = true
+                    redirect_stderr(Base.devnull) do
```

Comment:
Is `redirect_stderr` "concurrent-safe"?  I don't think `redirect_stderr` is safe to use with `@async` since it has to modify the global state including, IIUC, C-level file descriptor. (Unless it does magic that I don't know.)

Is this to make `Pkg.precompile` work well with <https://github.com/JuliaLang/julia/pull/37374>? If so, I think it'd be better to change the API of `compilecache` to something like `compilecache(pkg::PkgId, path::String, show_log_deps::Bool = true)`.

Reply:
No I didn't do this for compatibility with that PR, I did it because the try catch wasn't catching the error message from the julia sub process within compilecache, so even though the error wasn't throwing, stderr was making the log messy. 

Seems like the suggested change to compilecache would be a good idea. Should I PR it, or you?

Reply:
It'd be great if you can go ahead and do the PR. I don't have the time to do it right now.

Reply:
Just opened that PR https://github.com/JuliaLang/julia/pull/37596
This PR now uses that method arg to redirect stderr to devnull directly in the core pipeline